test/integration/listener_extension_discovery_integration_test.cc

```diff
@@ -731,3 +769,215 @@ TEST_P(ListenerExtensionDiscoveryIntegrationTest, TwoSubscriptionsConfigDumpWith
+  upstream_request_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "200"}}, true);
+  ASSERT_TRUE(response->waitForEndStream());
+  EXPECT_EQ("200", response->headers().getStatusValue());
+  log = waitForAccessLog(access_log_name_, 1);
+  EXPECT_THAT(log, testing::HasSubstr("200 \"def\""));
+  codec_client_->close();
+}
+
+TEST_P(QuicListenerExtensionDiscoveryIntegrationTest, BadEcdsUpdateWithoutDefault) {
+  on_server_init_function_ = [&]() { waitXdsStream(); };
+  // Do not have default listener filter config nor wait for ECDS update.
+  addDynamicFilter(/*apply_without_warming=*/false, /*set_default_config*/ false);
+  useAccessLog(fmt::format("%RESPONSE_CODE% %FILTER_STATE({})%",
+                           TestQuicListenerFilter::TestStringFilterState::key()));
+  initialize();
+
+  EXPECT_EQ(test_server_->server().initManager().state(), Init::Manager::State::Initializing);
+
+  // This is a bad config with empty value string.
+  sendXdsResponse(filter_name_, "v1", "");
+  test_server_->waitForCounterGe(
+      "extension_config_discovery.quic_listener_filter." + filter_name_ + ".config_fail", 1);
+  test_server_->waitForCounterGe(
+      "extension_config_discovery.quic_listener_filter." + filter_name_ + ".config_reload", 1);
+  test_server_->waitUntilListenersReady();
+  test_server_->waitForGaugeGe("listener_manager.workers_started", 1);
+  EXPECT_EQ(test_server_->server().initManager().state(), Init::Manager::State::Initialized);
+
+  Network::ClientConnectionPtr conn = makeClientConnection(lookupPort(port_name_));
+  std::shared_ptr<Upstream::MockClusterInfo> cluster{new NiceMock<Upstream::MockClusterInfo>()};
+  Upstream::HostDescriptionConstSharedPtr host_description{Upstream::makeTestHostDescription(
+      cluster, fmt::format("tcp://{}:80", Network::Test::getLoopbackAddressUrlString(version_)),
+      timeSystem())};
+  auto codec = std::make_unique<IntegrationCodecClient>(
+      *dispatcher_, random_, std::move(conn), host_description, downstream_protocol_, true);
+  EXPECT_TRUE(codec->disconnected());
+  EXPECT_EQ("QUIC_NO_ERROR with details: Closed by application with reason: no filter chain found",
+            codec->connection()->transportFailureReason());
+  // The extension_config_missing stats counter increases by 1.
+  test_server_->waitForCounterGe("listener.listener_stat.extension_config_missing", 1);
+
+  sendXdsResponse(filter_name_, "v2", "abc");
+  test_server_->waitForCounterGe(
+      "extension_config_discovery.quic_listener_filter." + filter_name_ + ".config_reload", 2);
+  testRouterHeaderOnlyRequestAndResponse();
+  codec_client_->close();
+  std::string log = waitForAccessLog(access_log_name_, 0);
+  EXPECT_THAT(log, testing::HasSubstr("200 \"abc\""));
+}
+
+TEST_P(QuicListenerExtensionDiscoveryIntegrationTest, ConfigDump) {
+  DISABLE_IF_ADMIN_DISABLED; // Uses admin interface.
+  on_server_init_function_ = [&]() { waitXdsStream(); };
+  addDynamicFilter(/*apply_without_warming=*/false);
```

Comment:
For configDump, could you please add two TCP listener filters and two QUIC listener filters, and see how the config dump looks like?